Automattic/wp-calypso
# Enhance/display important dates on search

#### Changes proposed in this Pull Request

* Depending on post/page status, display the most meaningful date
* On search, display the said date with the status label
* ~Small~ Good Refactor of `PostRelativeTime`

 Search results should look close to this mockup:
![Screenshot 2020-04-15 at 00 31 49](https://user-images.githubusercontent.com/790558/79280455-b1c45780-7eb0-11ea-979a-84a6532dfa36.png)




#### Testing instructions



* Go to pages or posts views. Make sure displayed dates are the right ones
* When searching, make sure the dates are the same and label is displayed



#### Extras
Found a bug reported here
https://github.com/Automattic/wp-calypso/issues/41245
As it is out of scope for this one.


Fixes #41075

cc @mmtr @sixhours
